tutor/src/components/annotations/annotation.jsx

```diff
@@ -0,0 +1,373 @@
+      this.needToLoadAnnotations = false;
+    }
+  }
+
+  @action.bound
+  highlightAndClose() {
+    return this.saveAnnotation().then(
+      action((response) => {
+        this.widgetStyle = null;
+        this.highlightEntry(response);
+        return response;
+      }));
+  }
+
+  @autobind
+  openAnnotator() {
+    return this.highlightAndClose().then(
+      action((response) => {
+        this.activeHighlight = response;
+      }));
+  }
+
+  @action.bound
+  updateActiveAnnotation(event) {
+    const newValue = event.target.value;
+
+    this.activeHighlight = Object.assign(
+      {},
+      this.activeHighlight,
+      {annotation: newValue}
+    );
+  }
+
+  @action.bound
+  updateAnnotation() {
+    const oldEntry = this.preload.find((e) => e.savedId === this.activeHighlight.savedId);
+
+    this.props.windowImpl.getSelection().empty();
+    if (oldEntry.annotation !== this.activeHighlight.annotation) {
+      hypothesisStore.update(this.activeHighlight.savedId, this.activeHighlight.annotation).then(
+        action((response) => {
+          const oldEntry = this.preload.find((e) => e.savedId === response.id);
+
+          Object.assign(oldEntry, dbEntryToPreload(response));
+          return oldEntry;
+        }));
+    }
+    this.activeHighlight = null;
+  }
+
+  @action.bound
+  saveAnnotation() {
+    this.props.windowImpl.getSelection().empty();
+
+    return hypothesisStore.save(this.props.documentId, this.savedSelection, '')
+      .then(action((response) => {
+        const newEntry = dbEntryToPreload(response);
+        const newPreload = this.preload.slice();
+
+        newPreload.push(newEntry);
+        this.preload = newPreload;
+        return newEntry;
+      }));
+  }
+
+  @action.bound
+  deleteEntry() {
+    this.unhighlightEntry(this.activeHighlight);
+    hypothesisStore.delete(this.activeHighlight.savedId).then(
+      action((response) => {
+        const oldIndex = this.preload.findIndex((e) => e.savedId === response.id);
+        const newPreload = this.preload.slice();
+
+        newPreload.splice(oldIndex, 1);
+        this.preload = newPreload;
+      }));
+    this.activeHighlight = null;
+    this.props.windowImpl.getSelection().empty();
+  }
+
+  @action
+  nextAnnotationInSortedList(entries) {
+    const nextIndex = 1 + entries.findIndex((e) => e.selection.start === this.activeHighlight.selection.start);
+
+    if (nextIndex < entries.length) {
+      this.activeHighlight = entries[nextIndex];
+      serializeSelection.restore(entries[nextIndex].selection, this.articleElement);
+    }
+  }
+
+  @autobind
+  nextAnnotation() {
+    const entries = this.preload.sort((a, b) => a.selection.start - b.selection.start);
+
+    this.nextAnnotationInSortedList(entries);
+  }
+
+  @autobind
+  previousAnnotation() {
+    // The trick is that the sort is reversed
+    const entries = this.preload.sort((a, b) => b.selection.start - a.selection.start);
+
+    this.nextAnnotationInSortedList(entries);
+  }
```

Comment:
I think this is missing a `return`

Reply:
No, these are onClick callbacks. They just move the highlight.